packages/@react-spectrum/s2/src/Disclosure.tsx

```diff
@@ -208,28 +260,50 @@ function DisclosureHeader(props: DisclosureHeaderProps, ref: DOMRef<HTMLDivEleme
   let {isExpanded} = useContext(DisclosureStateContext)!;
   let {size, density, isQuiet} = useSlottedContext(DisclosureContext)!;
   let isRTL = direction === 'rtl';
+
+  let ctx = useContext(InternalDisclosureHeader);
+  if (ctx) {
```

Comment:
The <Heading... /> look the same. I've looked twice and not seen a difference. Couldn't we do the following?
```
let heading = (<Heading ...>...</Heading>);
if (ctx) {
  return heading;
}

return (<<DisclosureHeader>{heading}</Heading>);
```

Reply:
yeah for whatever reason i couldn't get it to work but i think i was just missing something. i'll push up the changes soon!